test/modules/TestResultCoordinator/Reports/TestResultFilter.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestResultCoordinator
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Extensions.Logging;
+    using TestResultCoordinator.Reports;
+
+    /// <summary>
+    /// Longhaul tests need to filter out recent results, so recent unmatched results
+    /// that will soon be matched don't result in false failures.
+    ///
+    /// This abstraction filters out any result with a CreatedAt
+    /// within the configurable ignore threshold, whith
+    /// the exception of actual results matching a
+    /// non-filtered expected result.
+    /// </summary>
```

Comment:
If there is an unexpected actual result at the very beginning, followed by expected/actual pairs, this will lead to bad filtering. But this implementation is consistent with our report generators which suffer from the same issue. I have never seen this happen, and don't expect it to be possible.